docs/upgrade.html

```diff
@@ -19,6 +19,16 @@
 
 <script id="upgrade-template" type="text/x-handlebars-template">
 
+<h4><a id="upgrade_3_5_0" href="#upgrade_3_5_0">Upgrading to 3.5.0 from any version 0.8.x through 3.4.x</a></h4>
+
+    <h5><a id="upgrade_350_notable" href="#upgrade_350_notable">Notable changes in 3.5.0</a></h5>
+    <ul>
+        <li>The JmxTool has been migrated to the tools module.
+            As there is no wrapper script, users have to update their commands from "kafka-run-class.sh kafka.tools.JmxTool" to "kafka-run-class.sh org.apache.kafka.tools.JmxTool".
```

Comment:
It is a tad annoying that we are not backward compatible here. We usually never break compatibility without a deprecation period. Would it be imaginale to keep the old class name and to just call the new one? We could add a warning in the mean time and we could remove it in the next major release. This is just a thought.

Reply:
Hi David, thanks for raising this. I agree, even if this is not one of the most popular tool and there has been some discussion about replacement, I've recently seen some people using it for debugging purpose and maybe for automation.

I updated the relevant note with the deprecation plan you suggested and changed `kafka-run-class` scripts so that we can print a warning and redirect to the new implementation class. There are other tools with no wrapper script and we can simply add an if for each of them. Let me know what you think of this approach.

Reply:
Handling it in `kafka-run-class` does not seem to be the right way. Have you considered keeping the old class and using it to log a warning and to redirect?

Reply:
Yes I have and it's easier, but that would require adding tools dependency to core module and I had the impression we wanted to avoid that.

Reply:
For the sake of considering all ideas, if we think renaming these classes is problematic, another option would be to have Java classes in the tools module that match the old Scala classes name and package. This should allow existing commands to keep working while not requiring core to depend on tools.

I wonder if we should add scripts for all tools. That would separate the actual classes (which are not part of the public APIs) from the scripts which effectively are part of Kafka's public APIs.

Reply:
> For the sake of considering all ideas, if we think renaming these classes is problematic, another option would be to have Java classes in the tools module that match the old Scala classes name and package. This should allow existing commands to keep working while not requiring core to depend on tools.

I was thinking about the same. Then, we could add a warning in that class and remove it in the next major release.

Reply:
> For the sake of considering all ideas, if we think renaming these classes is problematic, another option would be to have Java classes in the tools module that match the old Scala classes name and package. 

I also tried this. The problem with this idea is that checkstyle is only able to handle a single root package per module, so you end up with the following error.

```sh
[ant:checkstyle] [ERROR] /home/fvaleri/Documents/kafka/tools/src/main/java/kafka/tools/JmxTool.java:17:1: Import control file does not handle this package. [ImportControl]
```

Possible alternatives: 

- create `:tools:deprecated` sub module containing `kafka.tools` root package with redirect classes (this works fine)
- use reflection inside core module to redirect without depending on tools module (have to try this one)